server/admin/http/handler.go

```diff
@@ -394,3 +404,50 @@ func decodePeerID(id string, w http.ResponseWriter) (peer.ID, bool) {
 	}
 	return peerID, true
 }
+
+func healthCheckValueStore(h *adminHandler) (cause ValueStoreFailReason, err error) {
+	mhash, val, err := createValueStoreDatum()
+	if err != nil {
+		return VS_GEN_FAILURE, err
+	}
+
+	err = h.indexer.Put(*val, mhash)
+	if err != nil {
+		return VS_WRITE_FAILURE, err
+	}
+
+	defer func() {
+		if errorOnRemove := h.indexer.Remove(*val, mhash); errorOnRemove != nil {
+			cause = VS_REMOVE_FAILURE
+			err = errorOnRemove
+		}
+	}()
+
+	rval, present, err := h.indexer.Get(mhash)
+
+	if err != nil {
+		return VS_READ_FAILURE, err
+	}
+	if !present {
+		return VS_NO_DATA, err
+	}
+	if !val.Equal(rval[0]) {
+		return VS_READWRITE_MISMATCH, err
+	}
+
+	return "", nil
+}
+
+func createValueStoreDatum() (multihash.Multihash, *indexer.Value, error) {
+	mhash := multihash.Multihash("2DrjgbFdhNiSJghFWcQbzw6E8y4jU1Z7ZsWo3dJbYxwGTNFmAj")
+	provider, err := peer.Decode("12D3KooWBUNzpAz1Jfvnaag1nHBi6gbG5Q1BQm8kCfcpmt7bGKa6")
+	if err != nil {
+		return nil, nil, err
+	}
+	val := indexer.Value{
+		ProviderID:    provider,
+		ContextID:     []byte(mhash),
+		MetadataBytes: []byte("p-metadata"),
+	}
+	return mhash, &val, nil
+}
```

Comment:
Since this data is always the same, instead of decoding the peer ID and creating the value on every call, it would be better to create a static `indexer.Value`

```suggestion
var healthCheckMH multihash.Multihash
var healthCheckValue indexer.Value

func init() {
	provider, err := peer.Decode("12D3KooWBUNzpAz1Jfvnaag1nHBi6gbG5Q1BQm8kCfcpmt7bGKa6")
	if err != nil {
		panic(err.Error())
	}
	healthCheckMH = multihash.Multihash("2DrjgbFdhNiSJghFWcQbzw6E8y4jU1Z7ZsWo3dJbYxwGTNFmAj")
	healthCheckValue = indexer.Value{
		ProviderID:    provider,
		ContextID:     []byte(healthCheckMH),
		MetadataBytes: []byte("healthcheck-metadata"),
	}
}
```